contracts/contracts/strategies/BaseConvexMetaStrategy.sol

```diff
@@ -0,0 +1,218 @@
+            address(this)
+        );
+        ICurvePool curvePool = ICurvePool(platformAddress);
+        if (contractPTokens > 0) {
+            uint256 virtual_price = curvePool.get_virtual_price();
+            uint256 value = contractPTokens.mulTruncate(virtual_price);
+            balance += value;
+        }
+
+        /* We intentionally omit the metapoolLp tokens held by the metastrategyContract
+         * since the contract should never (except in the middle of deposit/withdrawal
+         * transaction) hold any amount of those tokens in normal operation. There
+         * could be tokens sent to it by a 3rd party and we decide to actively ignore
+         * those.
+         */
+        uint256 metapoolGaugePTokens = IRewardStaking(cvxRewardStakerAddress)
+            .balanceOf(address(this));
+
+        if (metapoolGaugePTokens > 0) {
+            uint256 value = metapoolGaugePTokens.mulTruncate(
+                metapool.get_virtual_price()
+            );
+            balance += value;
+        }
+
+        uint256 assetDecimals = Helpers.getDecimals(_asset);
+        balance = balance.scaleBy(assetDecimals, 18) / THREEPOOL_ASSET_COUNT;
+    }
+
+    function _approveBase() internal override {
+        IERC20 pToken = IERC20(pTokenAddress);
+        // 3Pool for LP token (required for removing liquidity)
+        pToken.safeApprove(platformAddress, 0);
+        pToken.safeApprove(platformAddress, type(uint256).max);
+        // Gauge for LP token
+        metapoolLPToken.safeApprove(cvxDepositorAddress, 0);
+        metapoolLPToken.safeApprove(cvxDepositorAddress, type(uint256).max);
+        // Metapool for LP token
+        pToken.safeApprove(address(metapool), 0);
+        pToken.safeApprove(address(metapool), type(uint256).max);
+        // Metapool for Metapool main token
+        metapoolMainToken.safeApprove(address(metapool), 0);
+        metapoolMainToken.safeApprove(address(metapool), type(uint256).max);
+    }
+
+    /**
+     * @dev Get the index of the coin
+     */
+    function _getMetapoolCoinIndex(address _asset)
+        internal
+        view
+        returns (uint128)
+    {
+        for (uint128 i = 0; i < 2; i++) {
+            if (metapoolAssets[i] == _asset) return i;
+        }
+        revert("Invalid Metapool asset");
+    }
+
+    /**
+     * @dev Sets max withdrawal slippage that is considered when removing
+     * liquidity from Metapools.
+     * @param _maxWithdrawalSlippage Max withdrawal slippage denominated in
+     *        wad (number with 18 decimals): 1e18 == 100%, 1e16 == 1%
+     *
+     * IMPORTANT Minimum maxWithdrawalSlippage should actually be 0.1% (1e15)
+     * for production usage. Contract allows as low value as 0% for confirming
+     * correct behavior in test suite.
+     */
+    function setMaxWithdrawalSlippage(uint256 _maxWithdrawalSlippage)
+        external
+        onlyVaultOrGovernorOrStrategist
+    {
+        require(
+            _maxWithdrawalSlippage <= 1e17,
```

Comment:
I'm concerned that require this will result in us being stuck in the pool in case of a depeg.

Reply:
yes agree, better to have more flexibility than this sanity check... fix: https://github.com/OriginProtocol/origin-dollar/pull/1002/commits/48011870bea50ef45b101f623dd5ddf139914c03

Reply:
Looks good.